src/main/java/org/jabref/gui/fieldeditors/EditorTextArea.java

```diff
@@ -65,4 +72,42 @@ public void populateContextMenu(ContextMenu contextMenu) {
     public void initialize(URL location, ResourceBundle resources) {
         // not needed
     }
+
+    /**
+     * Set pasteActionHandler variable to passed handler
+     * @param  handler an instance of PasteActionHandler that describes paste behavior
+     */
+    public void setPasteActionHandler(PasteActionHandler handler) {
+        Objects.requireNonNull(handler);
+        this.pasteActionHandler = handler;
+    }
+  
+    /**
+     *  Override javafx TextArea method applying TextArea.paste() and pasteActionHandler after
+     */
+    @Override
+    public void paste() {
+        super.paste();
+        pasteActionHandler.handle();
+    }
+  
+    /**
+     *  Interface presents user-described paste behaviour applying to paste method
+     */
+    @FunctionalInterface
+    public interface PasteActionHandler {
+
+        void handle();
+    }
+  
+    /**
+     * Empty interface implementation to do nothing external on paste method
+     */
+    private static class EmptyPasteHandler implements PasteActionHandler {
```

Comment:
You could add this simply as default method in the interface, so you won't need that emptyPastHandller I think
https://dzone.com/articles/interface-default-methods-java

Reply:
Thanks for your feedback. I will try to fix this.